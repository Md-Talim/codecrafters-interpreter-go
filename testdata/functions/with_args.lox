// Test Case 1
// Prints 76
fun f1(a) {
  print a;
}
f1(76);

// Test Case 2
// Prints 72
fun f3(a, b, c) { print a + b + c; }
f3(24, 24, 24);

// Test Case 3
fun f8(a, b, c, d, e, f, g, h) { print a - b + c * d + e - f + g - h; }
f8(51, 51, 51, 51, 51, 51, 51, 51);

// Test Case 4
fun calculateGrade(score, bonus) {
  var finalScore = score + bonus;

  if (finalScore >= 90) {
    print "A";
  } else if (finalScore >= 80) {
    print "B";
  } else if (finalScore >= 70) {
    print "C";
  } else if (finalScore >= 60) {
    print "D";
  } else {
    print "F";
  }
}

var score = 81;
var bonus = 3;
print "Grade for given score is:";
calculateGrade(score, bonus);