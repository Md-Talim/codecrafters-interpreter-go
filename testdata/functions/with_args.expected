76
72
2601
Grade for given score is:
B